dashboard/modules/job/job_agent.py

```diff
@@ -0,0 +1,97 @@
+import aiohttp
+from aiohttp.web import Request, Response
+import dataclasses
+import json
+import logging
+import traceback
+
+import ray.dashboard.optional_utils as optional_utils
+import ray.dashboard.utils as dashboard_utils
+from ray.dashboard.modules.job.common import (
+    JobSubmitRequest,
+    JobSubmitResponse,
+)
+from ray.dashboard.modules.job.job_manager import JobManager
+from ray.dashboard.modules.job.utils import parse_and_validate_request, find_job_by_ids
+
+
+routes = optional_utils.ClassMethodRouteTable
+logger = logging.getLogger(__name__)
+
+
+class JobAgent(dashboard_utils.DashboardAgentModule):
+    def __init__(self, dashboard_agent):
+        super().__init__(dashboard_agent)
+        self._job_manager = None
+        self._gcs_job_info_stub = None
+
+    @routes.post("/api/job_agent/jobs/")
+    @optional_utils.init_ray_and_catch_exceptions()
+    async def submit_job(self, req: Request) -> Response:
+        result = await parse_and_validate_request(req, JobSubmitRequest)
+        # Request parsing failed, returned with Response object.
+        if isinstance(result, Response):
+            return result
+        else:
+            submit_request = result
+
+        request_submission_id = submit_request.submission_id or submit_request.job_id
+        try:
+            submission_id = await self.get_job_manager().submit_job(
+                entrypoint=submit_request.entrypoint,
+                submission_id=request_submission_id,
+                runtime_env=submit_request.runtime_env,
+                metadata=submit_request.metadata,
+                _driver_on_current_node=False,
```

Comment:
@Catch-Bull since `_driver_on_current_node=False`, it looks like a job can be submitted, handled by the job agent on worker node A, and then the driver script and supervisor actor gets run on a different worker node B.  

How does that work with the job log API?  In https://github.com/ray-project/ray/pull/28288 it looks like the logs will be requested on the JobAgent on node A, and the JobAgent will ask its `job_manager` for the logs, but the `job_manager` only looks at local files on node A.

Reply:
We can access SupervisorActor from GCS(or DataSource.Actors) by its name, and we can get the address that the driver/SupervisorActor running on, then we can get log directly from JobAgent

Reply:
@architkulkarni you can find some detail int here: https://github.com/alipay/ray/blob/ab77e300bc89c11b359fbb9b58efe305d16ae668/dashboard/modules/job/tests/test_job_agent.py#L428

Reply:
Ah makes sense, thanks :) So right now that logic only appears in that test file, but in the final state, will the flow look like this?

1. User requests logs via HTTP or SDK 
2. job_head uses get_supervisor_actor_info/GCS to get the address of the driver's node 
3. job_head gets the log from the agent on that node using JobAgentSubmissionClient, and returns it to the user

Reply:
Yes, I plan to implement it like this.
I discussed with @SongGuyang, that some deployment solutions may cause users to be unable to access other nodes except the head node, so it must be forwarded through the head node instead of directly accessing the agent from the client.